src/status_im/ui/screens/add_new/models.cljs

```diff
@@ -4,16 +4,24 @@
             [status-im.i18n :as i18n]
             [status-im.ui.screens.navigation :as navigation]
             [status-im.utils.universal-links.core :as universal-links]
-            [status-im.utils.fx :as fx]))
+            [status-im.utils.fx :as fx]
+            [status-im.browser.core :as browser]
+            [status-im.constants :as constants]
+            [status-im.extensions.core :as extensions]))
+
+(defn- valid-url? [url]
+  (boolean (re-matches constants/regx-url url)))
 
 (fx/defn process-qr-code
   [cofx data]
   (if (spec/valid? :global/public-key data)
     (universal-links/handle-view-profile cofx data)
     (or (universal-links/handle-url cofx data)
-        {:utils/show-popup {:title      (i18n/label :t/unable-to-read-this-code)
-                            :content    (i18n/label :t/use-valid-qr-code {:data data})
-                            :on-dismiss #(re-frame/dispatch [:navigate-to-clean :home])}})))
+        (if (valid-url? data)
```

Comment:
Maybe use some `cond` instead of the mix of or and if? That would require a predicate to detect universal links.

Reply:
okay I will clean this up.